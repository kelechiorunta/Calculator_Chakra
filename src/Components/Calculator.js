import React from 'react'
import { useReducer } from 'react'
import { CalcReducer } from '../Modules/CalcReducer'
import { Box, Button, Heading, Text, Grid } from '@chakra-ui/react'

var operand = 0;

function Calculator() {
    const initialParam = {digit:"0", operator:"", operatorClicked:false}
    const [state, dispatch] = useReducer(CalcReducer, initialParam)
    

    const appendDigit = (e) =>{
        dispatch({
            type: "APPEND",
            payload: e.target.textContent
        })
    }
    const resetDigit = () =>{
        dispatch({
            type:"RESET"
        })
    }
    const deleteDigit = () =>{
        dispatch({
            type:"DELETE"
        })
    }
    const computation = (value) =>{
        if (state.operator==="+"){
            operand += value
            value = 0
        }
        else if (state.operator==="-"){
            operand -= value
            value = 0
        }
        else if (state.operator==="*"){
            operand *= value
        }
        else if (state.operator==="/"){
            operand /= value
        }
        else{
            operand = value
        }
        return operand
    }

    const getComputation = (e) =>{
        dispatch({
            type: "COMPUTE",
            result: (state.operatorClicked===false) ? computation(state.digit): state.digit,
            nextOperator: e.target.textContent,
            operatorClicked: true
        })
    }
  return (
    <Box className='container-grid' p={4} width={{base:'97%', sm:'97%', md:'95%', lg:'58%'}}
    height={{base:'97%', sm:'97%', md:'95%', lg:'98%'}}
    m={{base:'8px', sm:'12px', md:'15px', lg:'auto'}} shadow={'md'}
    border={'1px solid blue'} borderRadius='10px'>
        <Grid p={4} shadow='md' height={{base:'97%', sm:'97%', md:'95%', lg:'98%'}}
        gridTemplateColumns={{base:'repeat(4, 1fr)', sm:'repeat(4, 1fr)', md:'repeat(5, 1fr)'}}
         gap={2}>
            
            <Heading data-testid='display' p={2} shadow='lg' textAlign='right' boxShadow='2px 2px 4px blue' mb={4}
            border={'1px solid blue'} borderRadius='10px' gridColumn={{base:'span 4', sm:'span 4', md:'span 5'}}>{state.digit}</Heading>
            
            <Button colorScheme='red' gridColumn={{base:'span 2', sm:'span 2', md:'span 2'}} 
            onClick={resetDigit}>CE</Button>
            <Button colorScheme='red' gridColumn={{base:'span 2', sm:'span 2', md:'span 3'}}
            onClick={deleteDigit}>DEL</Button>

            <Button colorScheme='blue'  
            onClick={appendDigit}>1</Button>
            <Button colorScheme='blue'  
            onClick={appendDigit}>2</Button>
            <Button colorScheme='blue'  
            onClick={appendDigit}>3</Button>
            <Button colorScheme='blue'  
            onClick={appendDigit}>4</Button>
            <Button colorScheme='blue'  
            onClick={appendDigit}>5</Button>
            <Button colorScheme='blue'  
            onClick={appendDigit}>6</Button>
            <Button colorScheme='blue'  
            onClick={appendDigit}>7</Button>
            <Button colorScheme='blue'  
            onClick={appendDigit}>8</Button>
            <Button colorScheme='blue'  
            onClick={appendDigit}>9</Button>
            <Button colorScheme='blue'  
            onClick={appendDigit}>0</Button>
            
             <Button colorScheme='green'  
            onClick={getComputation}>+</Button>
             <Button colorScheme='green'  
            onClick={getComputation}>-</Button>
             <Button colorScheme='green'  
            onClick={getComputation}>*</Button>
             <Button colorScheme='green'  
            onClick={getComputation}>/</Button>
             <Button colorScheme='green'  
            onClick={getComputation}>=</Button>
        </Grid>
           </Box>
  )
}

export default Calculator