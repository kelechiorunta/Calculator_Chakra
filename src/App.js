import './App.css';
import { useState } from 'react';
import Calculator from './Components/Calculator';
import { ChakraProvider, extendTheme } from '@chakra-ui/react';
import { Box, Button, Heading, Text } from '@chakra-ui/react'

function App() {
  const [isShow, setShow] = useState(false)
  
  const handleClick = () => {
    setShow(!isShow)
  }
  return (
    <ChakraProvider>
      <div className="App">
        <button className='show' onClick={handleClick}>{isShow? 'Show' : 'Hide'}</button>
        <button>Increment</button>
      
        <Calculator />

    </div>
  </ChakraProvider>
  );
}

export default App;
