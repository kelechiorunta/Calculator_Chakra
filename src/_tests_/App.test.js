import React from 'react';
import App from '../App';
import { create } from 'react-test-renderer'
import { getByTestId, render, screen, fireEvent} from '@testing-library/react'
import userEvent from '@testing-library/user-event';
import Calculator from '../Components/Calculator';



describe("Test 3", ()=>{
    test("Confirm Calculator Btns", () =>{
        const { getByText, getByTestId, asFragment } = render(<Calculator/>)
        expect(getByText("1")).toBeInTheDocument()
        expect(getByText("2")).toBeInTheDocument()
        expect(getByText("3")).toBeInTheDocument()
        expect(getByText("4")).toBeInTheDocument()
        expect(getByTestId('display')).toBeInTheDocument()
        expect(getByTestId('display')).toHaveTextContent(0)
        // fireEvent.click(getByText("1"))
        // expect(getByTestId('display')).toHaveTextContent(1)
        expect(asFragment()).toMatchSnapshot()
    })
})