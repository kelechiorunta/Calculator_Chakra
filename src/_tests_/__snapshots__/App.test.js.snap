// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My first snapshot test testing buttons 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <button
      class="show"
    >
      Hide
    </button>
    <button>
      Increment
    </button>
    <div
      class="container-grid css-k1l722"
    >
      <div
        class="css-1o6t3t0"
      >
        <h2
          class="chakra-heading css-syg44k"
        >
          0
        </h2>
        <button
          class="chakra-button css-1uueag5"
          type="button"
        >
          CE
        </button>
        <button
          class="chakra-button css-1ad2x05"
          type="button"
        >
          DEL
        </button>
        <button
          class="chakra-button css-jut409"
          type="button"
        >
          1
        </button>
        <button
          class="chakra-button css-jut409"
          type="button"
        >
          2
        </button>
        <button
          class="chakra-button css-jut409"
          type="button"
        >
          3
        </button>
        <button
          class="chakra-button css-jut409"
          type="button"
        >
          4
        </button>
        <button
          class="chakra-button css-jut409"
          type="button"
        >
          5
        </button>
        <button
          class="chakra-button css-jut409"
          type="button"
        >
          6
        </button>
        <button
          class="chakra-button css-jut409"
          type="button"
        >
          7
        </button>
        <button
          class="chakra-button css-jut409"
          type="button"
        >
          8
        </button>
        <button
          class="chakra-button css-jut409"
          type="button"
        >
          9
        </button>
        <button
          class="chakra-button css-jut409"
          type="button"
        >
          0
        </button>
        <button
          class="chakra-button css-h211ee"
          type="button"
        >
          +
        </button>
        <button
          class="chakra-button css-h211ee"
          type="button"
        >
          -
        </button>
        <button
          class="chakra-button css-h211ee"
          type="button"
        >
          *
        </button>
        <button
          class="chakra-button css-h211ee"
          type="button"
        >
          /
        </button>
        <button
          class="chakra-button css-h211ee"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`Test 3 Confirm Calculator Btns 1`] = `
<DocumentFragment>
  <div
    class="container-grid css-1hm7hk9"
  >
    <div
      class="css-ua1pc9"
    >
      <h2
        class="chakra-heading css-bnwn42"
        data-testid="display"
      >
        0
      </h2>
      <button
        class="chakra-button css-19lwpln"
        type="button"
      >
        CE
      </button>
      <button
        class="chakra-button css-91n2j2"
        type="button"
      >
        DEL
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        1
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        2
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        3
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        4
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        5
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        6
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        7
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        8
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        9
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        0
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        +
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        -
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        *
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        /
      </button>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</DocumentFragment>
`;
